src/qcodes/instrument_drivers/rigol/Rigol_DG4000.py

```diff
@@ -460,7 +465,7 @@ def __init__(
             self.add_parameter(
                 ch + "harmonic_order",
                 get_cmd=source + "HARM:ORDE?",
-                get_parser=int,
+                get_parser=float,
```

Comment:
I don't understand why this change is needed. Is it possible to set the harmonic_order to a non integer value and if so the validator below would also need to be changed. The same question applies to sweep_step_below

Reply:
okey，I will have a try. the firmware version of rigol dg4162 I used is `00.01.08`.
Actually I have already tried to update, but failed because I don't have U disk which has a capacity less than 4GB...😢. I will buy a new one.

Reply:
It's wired, right?

Reply:
Hi, @jenshnielsen 
I have one question: the return values of commands are defined in the [RIGOL DG4000 programming manual](https://www.rigolna.com/products/waveform-generators/dg4000/).
What I modified is based on the manual and has also been tested by myself.
It is to say that your already tested DG4000 series devices do not follow the programming manual correctly. Am I right? 🧐

Reply:
@konosubakonoakua I think we should retain `get_parser=int` this means that the instrument will return `7.000000E+00` which qcodes will then convert to `7` which is the format that the instrument expects as input. This means that you can do

```
myinstr.harmonic_order.set(myinstr.harmonic_order.get()
```
E.g. the output and the input has the same format. 

There is no need for us to reveal to the end user that the instrument returns a float and takes an integer as input